go/glide/lib.h

```diff
@@ -78,3 +78,13 @@ void close_client(const void *client_ptr);
  * * The contained `error_message` must be able to be safely casted to a valid `CString` via `CString::from_raw`. See the safety documentation of [`std::ffi::CString::from_raw`](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw).
  */
 void free_connection_response(const struct ConnectionResponse *connection_response_ptr);
+
+/**
+ * Deallocates an error message `CString`.
+ *
+ * # Safety
+ *
+ * * `error_msg_ptr` must be able to be safely casted to a valid `CString` via `CString::from_raw`. See the safety documentation of [`std::ffi::CString::from_raw`](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.from_raw).
+ * * `error_msg_ptr` must not be null.
+ */
+void free_error(const char *error_msg_ptr);
```

Comment:
Why re-expose?

Reply:
I assumed that the failure callback would use it for the error it gets for commands. Not relevant for connection logic though, so I guess it can be reverted. Up to you, since you have all the changes in your branch now.